### DISCUSSION TITLE
Add record level permission system

### DISCUSSION THREAD

**Discussion by maxpatiiuk** [specify/specify7#5613] [Feature Requests]
- Need to see how much demand there is for this
- Need to evaluate how hard this would be to implement (one implementation idea is to set permissions for a query, and all records returned by that query would inherit the permissions from the query)

**Comment by specifysoftware** [comment]
This issue has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/is-it-possible-to-write-protect-locality-records/930/2

**Comment by maxpatiiuk** [comment]
From @carlosmbe:

> **Is your feature request related to a problem? Please describe.**
> After Discussing with a Theresa. It was noted that some users would like a way to make certain attachments private so that not everyone with read access can just see all the collection's Attachments bare and exposed.
>
> **Describe the solution you'd like**
>
> Checkboxes in attachments 
> - [ ] Is Private 
> **OR MAYBE** 
> - [ ] Is Sensitive
> 
> They should prevent people lacking certain permissions from seeing said attachment. Or maybe it prevents people with a specific role from seeing it. 
> 
> The second checkbox (is sensitive) seems ideal if for example they have a visiting professional and they give them a custom role that is limited to what the Collection Manager wants them to see.
> 
> For more info, I'd recommend talking to  Theresa.
>

**Comment by maxpatiiuk** [comment]
See also https://github.com/specify/specify7/issues/2298 (a potential duplicate of this issue)

**Comment by grantfitzsimmons** [comment]
We need to make sure individual fields from individual records can be hidden

**Requested By:** KU Fish, KU Mammals, and many, many others
